### DISCUSSION TITLE
PDF preview conversions on production server is not created

### DISCUSSION THREAD

**Discussion by mostafakram** [spatie/laravel-medialibrary#2301] [Q&A]
I am running my queue server on 
php: 7.3 
laravel: 7
spatie/laravel-medialibrary: 7

I use this conversion to create pdf image previews 

```php
$this->addMediaConversion('portfolio-preview')
    ->extractVideoFrameAtSecond($second)
    ->performOnCollections('portfolio')
    ->height(500)
    ->width(500);
```

And i made sure i have `Ghostscript 9.50`, `imagick module version => 3.4.4` and `"spatie/pdf-to-image": "^2.0"`

The issue is conversion works fine on my local environment and staging server environment. It creates the preview for images, PDF and videos but on production server pdfs are not generated only images and videos 

I have tried to run it not on queue too as `->nonQueued()` but same result no error occurs and conversion is not created 

I have checked my s3 as well and i can find all of the conversions/ folders in all media types except pdf 

Any idea what i might be doing wrong ? Or how can i make it work ? 

Thanks for your help in advance

**Comment by aessam13** [comment]
@freekmurze Can you help us in this issue please :pray:

**Comment by mostafakram** [comment]
While debugging i found that when running `php artisan medialibrary:regenerate --only-missing --only=portfolio-preview` the conversion is being created

**Reply by mostafakram** [reply]
But at any place i write 

```php
Artisan::call('medialibrary:regenerate', [
                '--only-missing' => true,
                '--only' => 'portfolio-preview',
                '--force' => true,
            ]);
```

The conversions is not created 

even when i did 
```php
$schedule->command(RegenerateCommand::class, ['--only-missing' => true,'--only' => 'portfolio-preview', '--force' => true])->everyMinute();
```

The only thing that worked for me so far is ruuning the artisan command directly through ssh or schedule the command through forger scheduler to run every minute and i believe this is a bad approach
